sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/JsonCarbonWriter.java

```diff
@@ -91,4 +102,39 @@ public void close() throws IOException {
       throw new IOException(e);
     }
   }
+
+  private void loadSingleFile(File file) throws IOException {
+    try {
+      Reader reader = SDKUtil.buildJsonReader(file);
+      JSONParser jsonParser = new JSONParser();
+      Object jsonRecord = jsonParser.parse(reader);
+      if (jsonRecord instanceof JSONArray) {
+        JSONArray jsonArray = (JSONArray) jsonRecord;
+        for (Object record : jsonArray) {
+          this.write(record.toString());
+        }
+      } else {
+        this.write(jsonRecord.toString());
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      throw new IOException(e.getMessage());
```

Comment:
whether reader close  when it throws  exception?

Reply:
closed now.

Reply:
Need to close stream in validateJsonFiles as well.  Please check for all the stream cases in this PR.